assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -268,7 +268,7 @@ export const PaymentMethodDataProvider = ( { children } ) => {
 	// When checkout is returned to idle, set payment status to pristine
 	// but only if payment status is already not finished.
 	useEffect( () => {
-		if ( ! currentStatus.isSuccessful ) {
```

Comment:
oooo yup, good catch, I'm pretty sure I had it that way in #2425 but I must have resolved a conflict wrong when I was rebasing (dependency seems to support that :) )